### PR TITLE
OSDOCS-2012 Draft start for WMCO 3.x release notes

### PR DESCRIPTION
**Pull request by EricPonvelle**
This ticket is a duplicate of the one that Cody started. Fixes #31682

Preview: https://deploy-preview-32545--osdocs.netlify.app/openshift-enterprise/latest/windows_containers/windows-containers-release-notes-3-x.html

Jira: https://issues.redhat.com/browse/OSDOCS-2012

### CODE REVIEW

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (jrvaldes) [inline windows_containers/windows-containers-release-notes-3-x.adoc:44]: If I'm not mistaken, in AWS with CustomVXLANPort, Windows Server 2004 (SAC) is supported.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (jrvaldes) [inline windows_containers/windows-containers-release-notes-3-x.adoc:50]: Windows Server Semi-Annual Channel (SAC): Windows Server 2004  includes the patch already.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (jrvaldes) [inline windows_containers/windows-containers-release-notes-3-x.adoc:59]: WMCO 3.x is only compatible...

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (jrvaldes) [inline windows_containers/windows-containers-release-notes-3-x.adoc:101]: Consider removing empty sections

Comment (mansikulkarni96) [reply windows_containers/windows-containers-release-notes-3-x.adoc:101]: We may need to discuss this and add to it.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-3-x.adoc:44]: Would that change the value to "Windows Server 2004 (SAC)?"

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-3-x.adoc:50]: So, does that mean that I don't need to note this?

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-3-x.adoc:101]: This is still a work in progress in my view, so if we have no known fixes we can remove later.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-3-x.adoc:75]: Definitely, but I guess my biggest question is why would a customer want to use this functionality. It can be very high level and just a single sentence. I'm not sure I understand it enough to answer that though.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (jrvaldes) [reply windows_containers/windows-containers-release-notes-3-x.adoc:50]: Probably, let's run it by the Team first. 

Pointed that out, because should be explicit that the manual installation of the patch KB4566782 is not required with Windows Server Semi-Annual Channel (SAC): Windows Server 2004.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (saifshaikh48) [reply windows_containers/windows-containers-release-notes-3-x.adoc:75]: Cluster autoscaling is a provider-agnostic solution to dynamic computing resource requirements. Cluster autoscaling automatically scales up or down the number of machines cluster to meet its current deployment needs, keeping workloads available and pruning unused resources when possible.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [reply windows_containers/windows-containers-release-notes-3-x.adoc:44]: Can't we just link to the [pre-requisites](https://docs.openshift.com/container-platform/4.7/windows_containers/understanding-windows-container-workloads.html#wmco-prerequisites_understanding-windows-container-workloads) which clear explains what is supported?

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [reply windows_containers/windows-containers-release-notes-3-x.adoc:50]: This needs to be included. Someone could potentially bring up a 2004 instance without this patch especially when creating a golden image on vSphere.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:48]: This should be addressed in https://github.com/openshift/openshift-docs/pull/32773. This has been available since WMCO 2.x

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:42]: Do we really need to call this out here? This has been the case WMCO 1.x but just that users have not been grokking it correctly.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:52]: I suggest something along the lines of:
> Previously, a load balancer service would become unstable when the backing Deployment had multiple Pods scheduled on different Windows nodes. This issue has now been fixed.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:58]: I suggest something along the lines of:
> Previously, WMCO was adding the public key annotation, windowsmachineconfig.openshift.io/pub-key-hash, to Linux nodes. This no longer happens.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:60]: I suggest something along the lines of:
>Previously, the kube-proxy service would crash upon the creation of a load balancer service when the backing Deployment had multiple Pods scheduled on different Windows nodes. This issue has now been fixed.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:64]: [This comment](https://github.com/openshift/openshift-docs/pull/34130#discussion_r661766906) applies here too

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:68]: Do you release note bugs that were found before releasing the product? This bug can never be seen in the field.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:66]: PS: This was fixed in 2.0.1. Do we typically call that out here also?

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this, @EricPonvelle. Please address my comments.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-3-x.adoc:42]: Since we merged in the updates for 3.x, it can't hurt to spell this out, and then leave it here.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-3-x.adoc:68]: If this won't be seen, we can remove it. It's up to the team what should be highlighted, and I think it's typically stuff that would intrigue users to upgrade.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:49]: I would clarify and say `Now, WMCO no longer adds an annotation to Linux nodes`. We still add them to Windows nodes.

**Review by aravindhp — COMMENTED** [review]
/hold

until 3.0.0 is released. 

/cc @TanyaShearonRH

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (TanyaShearonRH) [inline windows_containers/windows-containers-release-notes-3-x.adoc:50]: Small edit. "The" should be lowercased. Instead of, "Previously, **T**he load balancer...." use "Previously, **t**he load balancer.."

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (TanyaShearonRH) [inline windows_containers/windows-containers-release-notes-3-x.adoc:63]: The Known issues section is empty. Should a statement be added that there are no issues or should the heading be removed (depending on current pattern)?

File: windows_containers/windows-containers-release-notes-2-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-2-x.adoc:75]: This is a known issue for WMCO 3.0.0 also and should be called out in the 3.0.0 release notes.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:45]: I suggest saying `must not` rather than `may not`

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (aravindhp) [inline windows_containers/windows-containers-release-notes-3-x.adoc:1]: This needs a known issue section, where we should mention:
> Users running WMCO 2.0.2 and below will have to uninstall the operator and reinstall WMCO 3.0.0 to work around https://bugzilla.redhat.com/show_bug.cgi?id=1983153

cc: @mansikulkarni96

**Review by aravindhp — CHANGES_REQUESTED** [review]
Thanks for working on this @EricPonvelle. Please address my comments.

**Review by aravindhp — COMMENTED** [review]
/lgtm

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (jeana-redhat) [inline windows_containers/windows-containers-release-notes-3-x.adoc:62]:
```suggestion
=== RHSA-2021:3001 - Windows Container support for {product-title} security update
```
Use the Live ID rather than the temporary ID. [76465](https://errata.devel.redhat.com/advisory/76465) seems to be RHSA-2021:3001

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (jeana-redhat) [inline windows_containers/windows-containers-release-notes-3-x.adoc:64]: I'm a bit confused about this v2 content's inclusion, and the reference to a v2 advisory under a v3 advisory header.

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (jeana-redhat) [inline windows_containers/windows-containers-release-notes-3-x.adoc:69]:
```suggestion
* The file system graphs available in the web console do not display for Windows nodes. This is caused by changes in the file system queries. This will be fixed in a future release of WMCO. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1930347[*BZ#1930347*])
```

**Review by jeana-redhat — COMMENTED** [review]
A few small fixes and a couple questions and suggestions. Looking good :)

File: windows_containers/windows-containers-release-notes-3-x.adoc
Comment (EricPonvelle) [reply windows_containers/windows-containers-release-notes-3-x.adoc:64]: @aravindhp Should we keep this included?

Comment (aravindhp) [reply windows_containers/windows-containers-release-notes-3-x.adoc:64]: I will defer to whatever pattern the docs team has been following regarding maintaining advisories across versions.

Comment (jeana-redhat) [reply windows_containers/windows-containers-release-notes-3-x.adoc:64]: I think the main thing to consider is: Is this relevant to v3 users? If not, I would lean towards removing it. v2 users should be getting relevant info in the v2 Rel Notes. But I don't really know enough about WinC to make an educated statement on this case :)

Comment (aravindhp) [reply windows_containers/windows-containers-release-notes-3-x.adoc:64]: This is not relevant to v3 users given it was fixed in the v2 timeframe.

**Pull request merged**
